iib/web/api_v1.py

```diff
@@ -170,22 +171,44 @@ def get_build_logs(request_id):
     :raise Gone: if the logs for the build request have been removed due to expiration
     """
     request_log_dir = flask.current_app.config['IIB_REQUEST_LOGS_DIR']
-    if not request_log_dir:
+    s3_bucket_name = flask.current_app.config['IIB_AWS_S3_BUCKET_NAME']
+    if not s3_bucket_name and not request_log_dir:
         raise NotFound()
 
     request = Request.query.get_or_404(request_id)
-    log_file_path = os.path.join(request_log_dir, f'{request_id}.log')
-    if not os.path.exists(log_file_path):
+
+    finalized = request.state.state_name in RequestStateMapping.get_final_states()
+    if not finalized:
```

Comment:
With this we are changing behavior also for loading log files from localhost (VM), right?
We get logs only in final state.

Reply:
yes, that's correct